src/GitPrompt.ps1

```diff
@@ -154,66 +154,69 @@ class PoshGitTextSpan {
-
-    LocalDefaultStatusSymbol               = [PoshGitTextSpan]::new('',  [ConsoleColor]::DarkGreen)
-    LocalWorkingStatusSymbol               = [PoshGitTextSpan]::new('!', [ConsoleColor]::DarkRed)
-    LocalStagedStatusSymbol                = [PoshGitTextSpan]::new('~', [ConsoleColor]::DarkCyan)
-
-    BranchGoneStatusSymbol                 = [PoshGitTextSpan]::new([char]0x00D7, [ConsoleColor]::DarkCyan) # × Multiplication sign
-    BranchIdenticalStatusSymbol            = [PoshGitTextSpan]::new([char]0x2261, [ConsoleColor]::Cyan)     # ≡ Three horizontal lines
-    BranchAheadStatusSymbol                = [PoshGitTextSpan]::new([char]0x2191, [ConsoleColor]::Green)    # ↑ Up arrow
-    BranchBehindStatusSymbol               = [PoshGitTextSpan]::new([char]0x2193, [ConsoleColor]::Red)      # ↓ Down arrow
-    BranchBehindAndAheadStatusSymbol       = [PoshGitTextSpan]::new([char]0x2195, [ConsoleColor]::Yellow)   # ↕ Up & Down arrow
-
-    BeforeIndexText                        = [PoshGitTextSpan]::new('',  [ConsoleColor]::DarkGreen)
-    BeforeStashText                        = [PoshGitTextSpan]::new(' (', [ConsoleColor]::Red)
-    AfterStashText                         = [PoshGitTextSpan]::new(')',  [ConsoleColor]::Red)
-
-    FileAddedText                          = '+'
-    FileModifiedText                       = '~'
-    FileRemovedText                        = '-'
-    FileConflictedText                     = '!'
-    BranchUntrackedText                    = ''
-
-    BranchBehindAndAheadDisplay            = [BranchBehindAndAheadDisplayOptions]"Full"
-
-    EnableStashStatus                      = $false
-    ShowStatusWhenZero                     = $true
-    AutoRefreshIndex                       = $true
-
-    EnablePromptStatus                     = !$global:GitMissing
-    EnableFileStatus                       = $true
-    EnableFileStatusFromCache              = $null
-    RepositoriesInWhichToDisableFileStatus = @() # Array of repository paths
-    DescribeStyle                          = ''
-
-    EnableWindowTitle                      = 'posh~git ~ '
-
-    AnsiConsole                            = $Host.UI.SupportsVirtualTerminal -or ($Env:ConEmuANSI -eq "ON")
-
-    DefaultPromptPrefix                    = ''
-    DefaultPromptSuffix                    = '$(''>'' * ($nestedPromptLevel + 1)) '
-    DefaultPromptDebugSuffix               = ' [DBG]$(''>'' * ($nestedPromptLevel + 1)) '
-    DefaultPromptEnableTiming              = $false
-    DefaultPromptAbbreviateHomeDirectory   = $false
-
-    Debug                                  = $false
-
-    BranchNameLimit                        = 0
-    TruncatedBranchSuffix                  = '...'
+class GitPromptSettings {
+    [bool]$AnsiConsole = $Host.UI.SupportsVirtualTerminal -or ($Env:ConEmuANSI -eq "ON")
+
+    [PoshGitCellColor]$DefaultColor   = [PoshGitCellColor]::new()
+    [PoshGitCellColor]$BranchColor    = [PoshGitCellColor]::new([ConsoleColor]::Cyan)
```

Comment:
Are PowerShell classes able to define type coersion? Would be handy to be able to asign `[ConsoleColor]::Cyan` here and have it automatically convert to a foreground-only color.